### ISSUE TITLE
程序跑的过程中有的时候会突然出现问题

### ISSUE THREAD

**Issue by sugurs** [chencodeX/triplet-loss-pytorch#1]
您好，感谢您创建了一个这么好的项目。我将您的项目应用在了我的数据集上，确实获得了性能提升。但是有的时候跑着跑着会出现以下问题，而导致程序终止：

![image](https://user-images.githubusercontent.com/29123728/85504701-1b827f80-b61f-11ea-8577-a5379c80ecbd.png)

刚开始第一个epoch一般不会出现，每次跑基本上都会在某个epoch出现而终止。我基本上没改您的程序，期待您帮忙解决一下这个问题，感谢！（我的训练集有900张（A类173，B类727张），测试集379张（A类75张，B类304张），训练的batch_size是按照您设置的是10，测试的batch_size是100）

**Comment by chencodeX** [comment]
可能是dataloader有问题，我查一下

**Comment by sugurs** [comment]
十分感谢您迅速的回复！

**Comment by zhzaozao** [comment]
请问这个问题解决了吗，我遇到一样的情况，还会卡死在   def __next__(self):里第一个while len(self.data_queue) < self.batch_size:里

**Comment by chencodeX** [comment]
可以提供一下您的数据集的分布情况，以及您使用的一些参数设置吗？还有出问题时是第几个epoch

**Comment by zhzaozao** [comment]
我训大库的时候发现的问题，然后随便拿一个小库试也是一样的问题，（两类，每类48个样本），batchsize 10，出现卡死 。第一个epoch好像是不会，一般第3个。我把triplet准备部分去掉了测试的

def create_txtfea_db(self):
        self.txtfea_db = []
        anchor_db = copy.deepcopy(self.lable_package)
	
        anchor_db_keys = list(anchor_db.keys())
        for lable in anchor_db_keys:	   
            lable_samples_anchor = anchor_db[lable]        
            if len(lable_samples_anchor) == 0:
                continue
            lable_sample_anchor_keys = list(lable_samples_anchor.keys())
            if self.shuffle:
                random.shuffle(lable_sample_anchor_keys)
           
            mid_index = len(lable_sample_anchor_keys) // 2
            lable_sample_anchor_keys[mid_index], lable_sample_anchor_keys[-1] = lable_sample_anchor_keys[-1], lable_sample_anchor_keys[mid_index]              
	  
	    #end = time.time()
            for key_index in range(len(lable_sample_anchor_keys)):
                sample_anchor = lable_samples_anchor[lable_sample_anchor_keys[key_index]]
                triplet_sample = [sample_anchor]
                
                self.txtfea_db.append(triplet_sample)

	    #end = time.time()-end
   
            if self.shuffle:
                random.shuffle(self.txtfea_db)

**Comment by zhzaozao** [comment]
发生这个卡死的时候返回的len（train_loader）为0

**Comment by chencodeX** [comment]
我构造了一个小数据集，两类，每类48张，epoch 40，batch_size 10，dataloader正常运行，没有问题

**Comment by zhzaozao** [comment]
感谢，我再对一遍，跑你的测试代码试试，看着哪都对但就是运行不正常
